server/datastore/mysql/software.go

```diff
@@ -415,18 +415,14 @@ func selectSoftwareSQL(opts fleet.SoftwareListOptions) (string, []interface{}, e
 			"s.arch",
 			"scv.cpe_id", // for join on sub query
 			goqu.COALESCE(goqu.I("scp.cpe"), "").As("generated_cpe"),
+		).
+		Join( // filter software that is not associated with any hosts
+			goqu.I("host_software").As("hs"),
+			goqu.On(
+				goqu.I("hs.software_id").Eq(goqu.I("s.id")),
+			),
 		)
 
-	if opts.HostID != nil || opts.TeamID != nil {
```

Comment:
I can't fully understand why this was the condition to do the join 🤔

Reply:
gotcha, thanks for the explanation!

Reply:
If you want to only get software from a specific host (`opts.HostID != nil`) or only get software from hosts of a specific team (`opts.TeamID != nil`), you need to join `host_software` (and below you can see a few more joins needed for each case).

Reply:
Oh I just saw Michal's explanation, Github failed to reload that for me.

Reply:
thanks both!